### DISCUSSION TITLE
Add interfaces and configure ip

### DISCUSSION THREAD

**Discussion by rudiservo** [O-X-L/ansible-opnsense#100] [Q&A] [answered]
I have been checking and ATM I only find Vlan, VX_lan and VIP.
How do you add interfaces, configure the ip, set it's interface, prevent removal, etc?

**Comment by superstes** [comment, marked as answer]
The OPNSense product does not yet provide an API for managing them.
See: https://docs.opnsense.org/development/api/core/interfaces.html

**Reply by rudiservo** [reply]
Yeah, it does not, how weird...

**Reply by superstes** [reply]
From what I know, they are currently migrating services and functionalities bit by bit from their legacy system to the API based one.

**Comment by ji-podhead** [comment]
> I have been checking and ATM I only find Vlan, VX_lan and VIP. How do you add interfaces, configure the ip, set it's interface, prevent removal, etc?

check out my backend api. you can call the opnsense scripts (over 80) , pluginctl, configctl and use the config manager to use configd and the xml configuration. If you provide the required xml data needed, ill create the classes and add them to config_manager.
rn you can set interfaces, dhcpd and vlans via config_manager. so you can use them directly, but im currently writing the docs so, ansible collection will follow soon. 
note that this is only possible due to using the backend, since this also reconfigures your vlans, interfaces...

**Reply by superstes** [reply]
Yeah - as far as I know there is no REST API for it yet.
And handling the 'legacy' XML interfaces is not recommended by the opnsense maintainers and thus is out of scope for this project.
But if you want to do so there are some other community-collections that do so

**Reply by leo1ardo** [reply]
that was replied to almost exactly 365 days later xD

**Comment by iamsilk** [comment]
Looks like an API change for this is in OPNsense's roadmap for 26.1 (January 2026): https://opnsense.org/roadmap/

**Comment by superstes** [comment]
Will be implemented the next months after the API is available: https://github.com/O-X-L/ansible-opnsense/issues/357
